x-pack/plugins/code/server/lsp/proxy.ts

```diff
@@ -175,28 +176,32 @@ export class LanguageServerProxy implements ILanguageServerHandler {
   }
 
   public awaitServerConnection() {
-    return new Promise((res, rej) => {
-      const server = net.createServer(socket => {
-        this.initialized = false;
-        server.close();
-        this.eventEmitter.emit('connect');
-        socket.on('close', () => this.onSocketClosed());
+    if (!this.connectingPromise) {
```

Comment:
nit: so this tries to dedup the connection promises, right? can we leave some comment on here on what issue it solves.